src/eduid_common/api/app.py

```diff
@@ -48,20 +50,72 @@
 from eduid_common.api.middleware import PrefixMiddleware
 from eduid_common.api.request import Request
 from eduid_common.api.utils import init_template_functions
-from eduid_common.authn.middleware import AuthnApp
-from eduid_common.config.app import EduIDApp
 from eduid_common.config.base import FlaskConfig
 from eduid_common.config.exceptions import BadConfiguration
 from eduid_common.config.parsers.etcd import EtcdConfigParser
 from eduid_common.session.eduid_session import SessionFactory
-from eduid_common.stats import NoOpStats, Statsd
-from eduid_userdb import UserDB
+from eduid_common.stats import init_app_stats
 
 DEBUG = os.environ.get('EDUID_APP_DEBUG', False)
 if DEBUG:
     stderr.writelines('----- WARNING! EDUID_APP_DEBUG is enabled -----\n')
 
 
+class EduIDApp(Flask):
+
+    def __init__(self, name: str, config: Mapping = None, init_central_userdb: bool = True, **kwargs):
+        super(EduIDApp, self).__init__(name, **kwargs)
+        if config is None:
+            warnings.warn("config argument should be set to an app class config object",
+                          DeprecationWarning)
+            return
+        # Only try to load the key, value pairs that FlaskConfig expects
+        field_names = set(f.name for f in fields(FlaskConfig))
+        filtered_config = {k: v for k, v in config.items() if k in field_names}
+        # XXX: try except
+        self.config = FlaskConfig(**filtered_config)
+
+        if DEBUG:
+            init_app_debug(self)
+
+        # Check that SECRET_KEY is set
+        if not self.config.secret_key:
+            raise BadConfiguration('SECRET_KEY is missing')
+
+        # App setup
+        self.wsgi_app = ProxyFix(self.wsgi_app)  # type: ignore
+        self.request_class = Request
+        self.url_map.strict_slashes = False
+
+        # Set app url prefix to APPLICATION_ROOT
+        self.wsgi_app = PrefixMiddleware(self.wsgi_app, prefix=self.config.application_root,  # type: ignore
+                                         server_name=self.config.server_name)
+
+        # Initialize shared features
+        init_logging(self)
+        init_exception_handlers(self)
+        init_sentry(self)
+        init_template_functions(self)
+        init_app_stats(self)
+        self.session_interface = SessionFactory(asdict(self.config))
+
+        if init_central_userdb:
+            self.central_userdb = UserDB(self.config.mongo_uri, 'eduid_am')
+
+        # Set up generic health check views
+        from eduid_common.api.views.status import status_views
+        self.register_blueprint(status_views)
+
+    def init_config(self, config_class, config):
```

Comment:
Not sure we still need this, in my memory I had removed it. I am testing it now in master, removing the method from `eduid_common.config.app` (and its execution in `eduid_common.api.app`) and it seems that it's only needed to allow checking for the `secret_key`.
mypy was being placated in each webapp, with the module from which current_app was being imported...

Reply:
Ignore this, sorry, there's more breakage. I agree that this should go, but it needs a bit more research.

Reply:
The init_config can go when all apps are instantiated in the [app_name]_init_app function I think.